File: android/app/src/main/kotlin/com/example/uber_surge_map_hk/model/OrderModel.kt
package com.example.uber_surge_map_hk.model

import com.google.gson.Gson

enum class OrderResult {
    PENDING, AUTO_ACCEPTED, MANUAL_ACCEPTED, REJECTED, MISSED;

    fun label(): String = when (this) {
        PENDING       -> "待處理"
        AUTO_ACCEPTED -> "自動接單"
        MANUAL_ACCEPTED -> "手動接單"
        REJECTED      -> "已拒絕"
        MISSED        -> "已錯過"
    }
}

data class OrderModel(
    val id: String,
    val fare: Double,
    val tripDistanceKm: Double,
    val pickupDistanceKm: Double,
    val pickupAddress: String,
    val destinationAddress: String,
    val timestampMs: Long,
    var result: OrderResult = OrderResult.PENDING
) {
    val farePerKm: Double get() = if (tripDistanceKm > 0) fare / tripDistanceKm else 0.0

    companion object {
        private val gson = Gson()
        fun fromJson(json: String): OrderModel? = runCatching { gson.fromJson(json, OrderModel::class.java) }.getOrNull()
    }

    fun toJson(): String = Gson().toJson(this)
}

File: lib/models/order_model.dart
import 'dart:convert';

enum OrderResult { pending, autoAccepted, manualAccepted, rejected, missed }

class OrderModel {
  final String id;
  final double fare;
  final double tripDistanceKm;
  final double pickupDistanceKm;
  final String pickupAddress;
  final String destinationAddress;
  final DateTime timestamp;
  OrderResult result;

  OrderModel({
    required this.id,
    required this.fare,
    required this.tripDistanceKm,
    required this.pickupDistanceKm,
    required this.pickupAddress,
    required this.destinationAddress,
    required this.timestamp,
    this.result = OrderResult.pending,
  });

  double get farePerKm =>
      tripDistanceKm > 0 ? fare / tripDistanceKm : 0;

  factory OrderModel.fromJson(String jsonStr) {
    final map = json.decode(jsonStr) as Map<String, dynamic>;
    return OrderModel(
      id: '${map['timestamp'] ?? DateTime.now().millisecondsSinceEpoch}',
      fare: (map['fare'] as num?)?.toDouble() ?? 0,
      tripDistanceKm: (map['tripDistanceKm'] as num?)?.toDouble() ?? 0,
      pickupDistanceKm: (map['pickupDistanceKm'] as num?)?.toDouble() ?? 0,
      pickupAddress: map['pickupAddress'] as String? ?? '',
      destinationAddress: map['destinationAddress'] as String? ?? '',
      timestamp: DateTime.fromMillisecondsSinceEpoch(
        (map['timestamp'] as num?)?.toInt() ?? 0,
      ),
      result: map['event'] == 'accepted'
          ? OrderResult.autoAccepted
          : OrderResult.pending,
    );
  }

  Map<String, dynamic> toMap() => {
        'id': id,
        'fare': fare,
        'tripDistanceKm': tripDistanceKm,
        'pickupDistanceKm': pickupDistanceKm,
        'pickupAddress': pickupAddress,
        'destinationAddress': destinationAddress,
        'timestamp': timestamp.millisecondsSinceEpoch,
        'result': result.name,
      };

  String get resultLabel {
    switch (result) {
      case OrderResult.autoAccepted:
        return '自動接單';
      case OrderResult.manualAccepted:
        return '手動接單';
      case OrderResult.rejected:
        return '已拒絕';
      case OrderResult.missed:
        return '已錯過';
      case OrderResult.pending:
        return '待處理';
    }
  }
}
